scripts/services/check-removed-services.js

```diff
@@ -0,0 +1,128 @@
+ *
+ * @param {Promise<Array<object>>} jsonData - An array of objects from services.json file.
+ * @returns {Promise<Array<string>>} An array of normalized service id form JSON.
+ */
+const getJsonObjectNames = (jsonData) => jsonData.map(({ id }) => normalizeFileName(id)).sort();
+
+/**
+ * Gets the names of YML file from the services folder.
+ *
+ * @param {string} servicesFolderPath - The path to the folder containing YML service files.
+ * @returns {Promise<Array<string>>} An array of normalized yml file names.
+ */
+const getYmlFileNames = async (servicesFolderPath) => {
+    // get all dir names from services folder
+    const ymlFiles = await fs.readdir(servicesFolderPath);
+    // get the file names without its extension
+    const ymlFileNames = ymlFiles.map((ymlFile) => path.parse(ymlFile).name);
+    // format file names
+    const formattedServiceNames = ymlFileNames.map(normalizeFileName);
+    // return sorted array
+    return formattedServiceNames.sort();
+};
+
+/**
+ * Write removed services objects into YML files.
+ *
+ * @param {Array<object>} removedObjects - Array of objects that should be written in separate YML files.
+ */
+const writeRemovedServices = async (removedObjects) => {
+    if (removedObjects.length === 0) {
+        return;
+    }
+    const [removedObject, ...restObjects] = removedObjects;
+    await fs.writeFile(
+        path.join(`${servicesDir}/${removedObject.id}${YML_FILE_EXTENSION}`),
+        yaml.dump(removedObject, { lineWidth: -1 }),
+    );
+    if (removedObjects.length > 1) {
+        await writeRemovedServices(restObjects);
+    }
+};
+
+/**
+ * Rewrites YAML files for removed services.
+ *
+ * @param {Array<string>} removedServicesNames - An array of removed normalized service names.
+ * @param {Array<object>} jsonServicesData - An array of json data objects
+ * @throws {Error} If there is an error while rewriting file.
+ */
+// TODO: get rid of "id" inside the "yml" file and take "id" directly from the "yml" filename
+// to avoid checking when adding new files and exclude the possibility of typos
+const rewriteYMLFile = async (removedServicesNames, jsonServicesData) => {
+    try {
+        // Get services objects from the json data, that have been deleted in services folder.
+        const removedServicesObjects = removedServicesNames
+            .map((removedServiceName) => jsonServicesData
+                .find(({ id }) => normalizeFileName(id) === removedServiceName));
+        await writeRemovedServices(removedServicesObjects);
+    } catch (error) {
+        throw new Error('Error while rewriting file:', error);
+    }
+};
+
+/**
+ * Checks for removed services and rewrites YAML files if necessary.
+ *
+ * @param {string} servicesFolderPath - The path to the folder /services.
+ * @param {string} jsonFilePath - The path to the JSON file containing old services data /assets/services.json.
+ */
+const checkRemovedServices = async (servicesFolderPath, jsonFilePath) => {
```

Comment:
the function not just "checks the removed services", it actually restores ones if there are such. function names and descriptions should be as clear as possible. I suggest such changes (feel free to fix if something is wrong):
```
/**
 * Checks if any of the input service data objects is removed
 * and restores it from the `resultFilePath` file.
 *
 * IMPORTANT: Services which previously were built to the `resultFilePath` file **should not be removed**.
 *
 * During the process service `id`s are checked against normalized YML file names
 * and if there are any differences, the corresponding service YML files are rewritten.
 *
 * <params description>
 */
const restoreRemovedInputServices = async (...) => { ... }
```